e2e/dractions/actions.go

```diff
@@ -131,24 +151,44 @@ func Failover(ctx types.Context) error {
 // Update DRPC
 func Relocate(ctx types.Context) error {
 	d := ctx.Deployer()
-	if d.IsDiscovered() {
-		return RelocateDiscoveredApps(ctx)
-	}
-
 	managementNamespace := ctx.ManagementNamespace()
 	log := ctx.Logger()
+	name := ctx.Name()
 
-	log.Infof("Relocating workload in namespace %q", managementNamespace)
+	drpcName := name
+	client := util.Ctx.Hub.Client
+	drPolicyName := util.DefaultDRPolicyName
+
+	currentCluster, err := getCurrentCluster(client, managementNamespace, name)
+	if err != nil {
+		return err
+	}
+
+	drpolicy, err := util.GetDRPolicy(client, drPolicyName)
+	if err != nil {
+		return err
+	}
+
+	targetCluster, err := getTargetCluster(client, managementNamespace, drpcName, drpolicy)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Relocating workload from cluster %q to cluster %q", currentCluster, targetCluster)
 
-	return failoverRelocate(ctx, ramen.ActionRelocate, ramen.Relocated)
+	if d.IsDiscovered() {
+		return failoverRelocateDiscoveredApps(ctx, ramen.ActionRelocate, ramen.Relocated, currentCluster, targetCluster)
+	}
+
+	return failoverRelocate(ctx, ramen.ActionRelocate, ramen.Relocated, targetCluster)
```

Comment:
Same comments as for failover.

We have some duplication, failover and relocate logging is almost the same, but 2 copies is fine.